feature-libs/product/configurators/rulebased/components/attribute/product-card/configurator-attribute-product-card.component.spec.ts

```diff
@@ -0,0 +1,216 @@
+import { By } from '@angular/platform-browser';
+import { ItemCounterComponent } from '@spartacus/storefront';
+import { FormControl } from '@angular/forms';
+import { ReactiveFormsModule } from '@angular/forms';
+
+const mockQuantity = new FormControl(1);
+
+describe('ConfiguratorAttributeProductCardComponent', () => {
+  let component: ConfiguratorAttributeProductCardComponent;
+  let fixture: ComponentFixture<ConfiguratorAttributeProductCardComponent>;
+
+  const createImage = (url: string, altText: string): Configurator.Image => {
+    const image: Configurator.Image = {
+      url: url,
+      altText: altText,
+    };
+    return image;
+  };
+
+  const createValue = (
+    description: string,
+    images: Configurator.Image[],
+    quantity: number,
+    selected: boolean,
+    valueCode: string,
+    valueDisplay: string
+  ): Configurator.Value => {
+    const value: Configurator.Value = {
+      description,
+      images,
+      quantity,
+      selected,
+      valueCode,
+      valueDisplay,
+    };
+    return value;
+  };
+
+  beforeEach(
+    waitForAsync(() => {
+      TestBed.configureTestingModule({
+        imports: [
+          I18nTestingModule,
+          ReactiveFormsModule,
+          RouterTestingModule,
+          UrlTestingModule,
+        ],
+        declarations: [
+          ConfiguratorAttributeProductCardComponent,
+          ConfiguratorShowMoreComponent,
+          ItemCounterComponent,
+        ],
+      })
+        .overrideComponent(ConfiguratorAttributeProductCardComponent, {
+          set: {
+            changeDetection: ChangeDetectionStrategy.Default,
+          },
+        })
+        .compileComponents();
+    })
+  );
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(
+      ConfiguratorAttributeProductCardComponent
+    );
+    component = fixture.componentInstance;
+    component.quantity = mockQuantity;
+    component.disabledAction = false;
+    component.multiSelect = false;
+    component.product = createValue(
+      'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
+      [createImage('url', 'alt')],
+      1,
+      false,
+      '1111-2222',
+      'Lorem Ipsum Dolor'
+    );
+
+    spyOn(component, 'onHandleSelect').and.callThrough();
+    spyOn(component, 'onHandleDeselect').and.callThrough();
+
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should button be enabled when disabledAction = false & selected = false', () => {
```

Comment:
change the test name, to something less logic-related. instead of `.. when disabledAction = false...`, use `should button be enabled when user doesn't exist` (example, test name not related)

Reply:
renamed